skiko/src/commonMain/cpp/common/common_interop.cc

```diff
@@ -0,0 +1,74 @@
+#include "common.h"
+#include "src/utils/SkUTF.h"
+
+jlong packTwoInts(int32_t a, int32_t b) {
+    return (uint64_t (a) << 32) | b;
+}
+
+jlong packIPoint(SkIPoint p) {
+    return packTwoInts(p.fX, p.fY);
+}
+
+jlong packISize(SkISize p) {
+    return packTwoInts(p.fWidth, p.fHeight);
+}
+
+namespace skija {
+
+    namespace SamplingMode {
+        SkSamplingOptions unpack(jlong val) {
+            if (0x8000000000000000 & val) {
+                val = val & 0x7FFFFFFFFFFFFFFF;
```

Comment:
Better use union to avoid aliasing issues. Also not sure if I understand the logic here, we use upper bit to select between resamplers?

Reply:
This code has just been moved verbatim 
from `src/jvmMain/cpp/common/interop.cc` 
to `src/commonMain/cpp/common/common_interop.cc `.
The jvm jni code still depends on it.
I'd like to make whatever semantic changes as a separate fix in the bindings-- this fix is HUGE already.

Reply:
OK